source/common/router/router.cc

```diff
@@ -111,6 +111,15 @@ FilterUtility::finalTimeout(const RouteEntry& route, Http::HeaderMap& request_he
     request_headers.insertEnvoyExpectedRequestTimeoutMs().value(expected_timeout);
   }
 
+  // If we've configured max_grpc_timeout, override the grpc-timeout header with
```

Comment:
I think this might actually be worth a release note. I didn't realize we weren't setting this header before. I thought @jrajahalme did it but I just looked and he only added the ability to configure an alternate gRPC timeout. Do you mind adding one?